### PR TITLE
[profiler] Enhancement: Add C callback

### PR DESCRIPTION
**Pull request by iyzhang**
This PR adds a C callback as an option to report statistics from our profiler

### CODE REVIEW

File: include/demi/types.h
Comment (ppenna) [inline include/demi/types.h:1]: There is no need to touch these files.

File: src/rust/demikernel/libos/mod.rs [hunk 1]
```diff
@@ -101,7 +103,7 @@ pub enum LibOS {
 /// Associated functions for LibOS.
 impl LibOS {
     /// Instantiates a new LibOS.
-    pub fn new(libos_name: LibOSName) -> Result<Self, Fail> {
+    pub fn new(libos_name: LibOSName, _perf_callback: Option<demi_callback_t>) -> Result<Self, Fail> {
```
Comment (ppenna) [inline src/rust/demikernel/libos/mod.rs:106]: Remove underscore from `perf_callback`, as this variable is being used.

**Review by ppenna — COMMENTED** [review]
LGTM

**Review by kyleholohan — APPROVED** [review]
lgtm

**Pull request merged**